ballerina/tests/test.bal

```diff
@@ -0,0 +1,249 @@
+    functions: [publicActionFunction],
+    labels: {
+        "en": {
+            "inputFieldLabels": {
+                "staticInput": "Static Input",
+                "objectInput": "Object Property Input",
+                "optionsInput": "External Options Input"
+            },
+            "actionName": "My Extension",
+            "actionDescription": "My Extension Description",
+            "appDisplayName": "My App Display Name",
+            "actionCardContent": "My Action Card Content"
+        }
+    }
+};
+
+
+
+#create Extension definition
+# 
+# + return - error? if an error occurs, null otherwise
+#
+@test:Config{groups: ["apikey"]}
+function testPost() returns error? {
+    io:println("Testing extension creation (POST)");
+    PublicActionDefinition response = check automationClient->/automation/v4/actions/[appId].post(testingPublicActionDefinitionEgg);
+
+    // Assert creation success and set the global ID
+    test:assertTrue(response?.id is string, "Extension creation failed");
+    io:println("Extension created successfully with ID: ", response?.id);
+    createdExtensionId = response.id;
+}
+
+
+# Insert a function for a definition
+# 
+# + return - error? if an error occurs, null otherwise
+# 
+@test:Config{
+    groups: ["apikey"],
+    dependsOn: [testPost]
+}
+ function testPostFunction() returns error? {
+    PublicActionFunctionIdentifier response = check automationClient->/automation/v4/actions/[appId]/[createdExtensionId]/functions/["POST_FETCH_OPTIONS"].put("exports.main = (event, callback) => {\r\n  callback({\r\n    \"options\": [{\r\n        \"label\": \"Big Widget\",\r\n        \"description\": \"Big Widget\",\r\n        \"value\": \"10\"\r\n      },\r\n      {\r\n        \"label\": \"Small Widget\",\r\n        \"description\": \"Small Widget\",\r\n        \"value\": \"1\"\r\n      }\r\n    ]\r\n  });\r\n}");
+}
+
+@test:Config {
+    dependsOn: [testPost],
+    groups: ["apikey"]
+}
+function testGetDefinitionById() returns error? {
+    io:println("Requesting extension by ID (GET)");
+    PublicActionDefinition response = check automationClient->/automation/v4/actions/[appId]/[createdExtensionId];
+
+    // Validate the retrieved extension's ID
+    test:assertTrue(response?.id === createdExtensionId, "Extension retrieval failed");
+    io:println("Extension retrieved successfully with ID: ", createdExtensionId);
+}
+
+
+
+# Get all functions for a given definition
+# 
+# + return - error? if an error occurs, null otherwise
+
+@test:Config{
+    dependsOn: [testPostFunction],
+    groups: ["apikey"]}
+ function testGetAllFunctions() returns error? {
+    io:println("requesting get all functions");
+    CollectionResponsePublicActionFunctionIdentifierNoPaging response = check automationClient->/automation/v4/actions/[appId]/[createdExtensionId]/functions;
+    //validate the response
+    // test:assertTrue(response?.results.length() > 0, "No functions found for the extension");
```

Comment:
Any reason to comment out the assertion ?

Reply:
no specific reason . it was failing in tests but then now sorted the issue. now uncommented -https://github.com/ballerina-platform/module-ballerinax-hubspot.automation.actions/pull/2/commits/acbce5b6cc9aaec87ffb6965c63cb150edbe13ab